### PR TITLE
Changed spin x taste structure for both the connected and disconnected diagrams

### PR DESCRIPTION
**Pull request by roberto-dionisio**

### CODE REVIEW

File: src/measures/fermions/ellesettete.cpp [hunk 1]
```diff
@@ -101,27 +103,35 @@ namespace nissa
...
-		    MINV(SEQ_PROP,iflav,SIMPLE_PROP);
-		    
-		    //then glb reduction to compute the trace for the connected 3pts diagram and the 2pts
-		    SUMM_THE_TIME_TRACE_PRINT_AT_LAST_HIT(Tr_two_pts,SIMPLE_PROP,SIMPLE_PROP);
+		    //compute  2pts propagator with id x g5 at source and apply id x g5 at sink
+		    MINV(ID_G5_PROP,iflav,id_g5_source);
+		    apply_stag_op(ID_G5_PROP,conf,theory_pars.backfield[iflav],GAMMA_INT::IDENTITY,GAMMA_INT::GAMMA_5,ID_G5_PROP);
+
+		    //compute sequential propagator with id x g5 at source and apply id x g5 at sink
+		    MINV(SEQ_PROP,iflav,ID_G5_PROP);
```
Comment (sunpho84) [inline src/measures/fermions/ellesettete.cpp:119]: The sequential propagator must have 1x1 in the insertion (mass term), so you should do it before line 116 (or better, use another structure to hold the result of line 116

File: src/measures/fermions/ellesettete.cpp [hunk 2]
```diff
@@ -101,27 +103,35 @@ namespace nissa
...
+		    apply_stag_op(SEQ_PROP,conf,theory_pars.backfield[iflav],GAMMA_INT::IDENTITY,GAMMA_INT::GAMMA_5,SEQ_PROP);
+
+		    //then glb reduction to compute the trace for the connected 2pts, 3pts and 4pts diagrams
+		    SUMM_THE_TIME_TRACE_PRINT_AT_LAST_HIT(Tr_two_pts,SIMPLE_PROP,ID_G5_PROP);
```
Comment (sunpho84) [inline src/measures/fermions/ellesettete.cpp:123]: ok must be done after the line 116 indeed

File: src/measures/fermions/ellesettete.cpp [hunk 3]
```diff
@@ -101,27 +103,35 @@ namespace nissa
...
+		    apply_stag_op(id_g5_source,conf,theory_pars.backfield[iflav],GAMMA_INT::IDENTITY,GAMMA_INT::GAMMA_5,source);
+
 		    //compute std 2pts propagator G(m|n) ~ [D^-1(m|y) source(y)] source(n)*
 		    MINV(SIMPLE_PROP,iflav,source);
 		    
-		    //compute sequential propagator G(m|n) ~ [D^-1(m|y) source(y)] D^-1(y|n)
-		    MINV(SEQ_PROP,iflav,SIMPLE_PROP);
-		    
-		    //then glb reduction to compute the trace for the connected 3pts diagram and the 2pts
-		    SUMM_THE_TIME_TRACE_PRINT_AT_LAST_HIT(Tr_two_pts,SIMPLE_PROP,SIMPLE_PROP);
```
Comment (sunpho84) [inline src/measures/fermions/ellesettete.cpp:114]: you should keep this

File: src/measures/fermions/ellesettete.cpp [hunk 4]
```diff
@@ -101,27 +103,35 @@ namespace nissa
...
+
+		    //compute sequential propagator with id x g5 at source and apply id x g5 at sink
+		    MINV(SEQ_PROP,iflav,ID_G5_PROP);
+		    apply_stag_op(SEQ_PROP,conf,theory_pars.backfield[iflav],GAMMA_INT::IDENTITY,GAMMA_INT::GAMMA_5,SEQ_PROP);
+
+		    //then glb reduction to compute the trace for the connected 2pts, 3pts and 4pts diagrams
+		    SUMM_THE_TIME_TRACE_PRINT_AT_LAST_HIT(Tr_two_pts,SIMPLE_PROP,ID_G5_PROP);
 		    SUMM_THE_TIME_TRACE_PRINT_AT_LAST_HIT(Tr_three_pts,SIMPLE_PROP,SEQ_PROP);
+		    apply_stag_op(SEQ_PROP,conf,theory_pars.backfield[iflav],GAMMA_INT::IDENTITY,GAMMA_INT::GAMMA_5,SEQ_PROP); //s.t. at sink we have back id x id
 		    SUMM_THE_TIME_TRACE_PRINT_AT_LAST_HIT(Tr_four_pts,SEQ_PROP,SEQ_PROP);
```
Comment (sunpho84) [inline src/measures/fermions/ellesettete.cpp:126]: NO as in this way, you are going to obtain g5 x g5 in both source and sink. Unfortunately, you need to create another sequential propagator using the plain source, such that when you close the two props you get the required spin/taste structure

Comment (sunpho84) [reply src/measures/fermions/ellesettete.cpp:126]: plus, you will need to adjust the sink according

File: src/measures/fermions/ellesettete.cpp [hunk 5]
```diff
@@ -101,27 +103,35 @@ namespace nissa
...
 
 		    //////// disconnected ////////
-		    apply_stag_op(source_g5_id,conf,theory_pars.backfield[iflav],GAMMA_INT::GAMMA_5,GAMMA_INT::IDENTITY,source);
-			
+		    //here we need a simple seq prop with nothing at source
+		    MINV(SEQ_PROP,iflav,SIMPLE_PROP);
```
Comment (sunpho84) [inline src/measures/fermions/ellesettete.cpp:130]: this is indeed the second sequential that you need for four pts of line 126

File: src/measures/fermions/ellesettete.cpp [hunk 6]
```diff
@@ -101,27 +103,35 @@ namespace nissa
...
 		    if(ihit==meas_pars.nhits-1) master_fprintf(file," # Tr_no_insertion_bubble source time %d\n", glb_t);
-		    SUMM_THE_TRACE_PRINT_AT_LAST_HIT(Tr_no_insertion_bubble,SIMPLE_PROP,source_g5_id);
+		    SUMM_THE_TRACE_PRINT_AT_LAST_HIT(Tr_no_insertion_bubble,SIMPLE_PROP,g5_id_source);
```
Comment (sunpho84) [inline src/measures/fermions/ellesettete.cpp:132]: for consistency, put the source as a second argument, as it should be eta^\dagger prop

File: src/measures/fermions/ellesettete.cpp [hunk 7]
```diff
@@ -101,27 +103,35 @@ namespace nissa
...
 		    if(ihit==meas_pars.nhits-1) master_fprintf(file,"\n # Tr_insertion_bubble source time %d\n", glb_t);
-		    SUMM_THE_TRACE_PRINT_AT_LAST_HIT(Tr_insertion_bubble,SEQ_PROP,source_g5_id);
+		    SUMM_THE_TRACE_PRINT_AT_LAST_HIT(Tr_insertion_bubble,SEQ_PROP,g5_id_source);
```
Comment (sunpho84) [inline src/measures/fermions/ellesettete.cpp:134]: similar here

**Review by sunpho84 — CHANGES_REQUESTED** [review]

File: src/measures/fermions/ellesettete.cpp [hunk 8]
```diff
@@ -101,27 +108,36 @@ namespace nissa
...
-		    SUMM_THE_TIME_TRACE_PRINT_AT_LAST_HIT(Tr_four_pts,SEQ_PROP,SEQ_PROP);
 
-		    //////// disconnected ////////
-		    apply_stag_op(source_g5_id,conf,theory_pars.backfield[iflav],GAMMA_INT::GAMMA_5,GAMMA_INT::IDENTITY,source);
+		    //compute  2pts propagator with id x g5 at source and apply id x g5 at sink
+		    MINV(PROP_ID_G5,iflav,id_g5_source);
+		    apply_stag_op(ID_G5_PROP_ID_G5,conf,theory_pars.backfield[iflav],GAMMA_INT::IDENTITY,GAMMA_INT::GAMMA_5,PROP_ID_G5);
+
+		    //compute sequential propagator with id x g5 at source and apply id x g5 at sink
+		    MINV(SEQ_PROP_ID_G5,iflav,ID_G5_PROP);
```
Comment (sunpho84) [inline src/measures/fermions/ellesettete.cpp:125]: no... here you are putting the insertion of the identity over spin and g5 over taste, but the sequential must have identity over taste

**Review by sunpho84 — CHANGES_REQUESTED** [review]

File: src/measures/fermions/ellesettete.cpp (hunk 8 above)
Comment (roberto-dionisio) [reply src/measures/fermions/ellesettete.cpp]: it should have been PROP_ID_G5 but I mispelled the name of the prop, which indeed doesn't exist in the code.

**Pull request merged**